### PR TITLE
Add Multi-select functionality to regime filter

### PR DESCRIPTION
**Pull request by antroy-madetech**
### Aim
Add Multi-select functionality to regime filter

[LTD-3829](https://uktrade.atlassian.net/browse/LTD-3829)

[LTD-3829]: https://uktrade.atlassian.net/browse/LTD-3829?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: caseworker/bookmarks/services.py [hunk 1]
```diff
@@ -57,56 +57,109 @@ def rename_bookmark(request, bookmark_id, name):
     return client.put(request, f"/bookmarks", data)
 
 
-def enrich_bookmark_for_display(bookmark, filter_data, all_flags):
```
Comment (currycoder) [inline caseworker/bookmarks/services.py:60]: FYI one of us will end up with a conflict; https://github.com/uktrade/lite-frontend/pull/1340

Comment (antroy-madetech) [reply caseworker/bookmarks/services.py:60]: OK - already had to resolve conflicts a few times in here.

File: caseworker/bookmarks/services.py [hunk 2]
```diff
@@ -57,56 +57,109 @@ def rename_bookmark(request, bookmark_id, name):
...
-    _enrich_value_from_filter_data(filter_dict, filter_data, "final_advice_type", "advice_types")
-
-    _swap_ids_for_readable_values(filter_dict)
-
-    _change_flag_ids_to_comma_seperated_names(filter_dict, all_flags)
-
-    return ", ".join(
-        [f"{k.capitalize().replace('_', ' ')}: {str(v).replace('_', ' ')}" for (k, v) in filter_dict.items()]
-    )
+class BookmarkEnricher:
```
[the remark this answers is not in the document]
Comment (antroy-madetech) [reply caseworker/bookmarks/services.py:60]: It's starting to get messy and passing the same bunch of params around to multiple functions. Can pull the functions back out if you want. Been thinking about 3855 and that was a driver for this. It feels like by the time we get to that ticket, we'll already have done all the enrichment we need for the foreseeable, and we may not even get to it, so was trying to get things tidied up for the tickets remaining.

File: caseworker/assets/javascripts/case-filters.js [hunk 3]
```diff
@@ -65,65 +64,27 @@ export default function initFlagsFiltersField() {
...
-function showHideFilters() {
-  let expandBasic = filterIsPopulated("basic-filter-fields");
-  let expandAdvanced = filterIsPopulated("advanced-filter-fields");
+function initRegimeFiltersField() {
+  const regimeField = document.getElementById("regime_entry");
+  if (!regimeField) return;
 
-  if (expandBasic || expandAdvanced) expandBasicFilters();
-  if (expandAdvanced) expandAdvancedFilters();
+  progressivelyEnhanceMultipleSelectField(regimeField, (option) => {
```
[the remark this answers is not in the document]
Comment (antroy-madetech) [reply caseworker/assets/javascripts/case-filters.js:85]: Nowhere near as many as cle. 667. But in any case they should probably use the same solution (unless that solution is freeform text field I guess)

File: caseworker/bookmarks/services.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (currycoder) [reply caseworker/bookmarks/services.py:60]: Ok fair enough.  I guess we can see how we get on as to whether we can do that ticket or not.

File: caseworker/assets/javascripts/case-filters.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (currycoder) [reply caseworker/assets/javascripts/case-filters.js:85]: OK cool, I'll hold off my :heavy_check_mark: until we get to the bottom of that

**Comment by antroy-madetech** [comment]
Merged with CLE work in a different branch

**Pull request closed without merging**